cognite/replicator/datapoints.py

```diff
@@ -1,28 +1,124 @@
+
+
+def _get_chunk(lst: List[Any], num_chunks: int, chunk_number: int) -> List[Any]:
+    """Returns a slice of the given list such that all slices are as even in size as possible.
+
+    Args:
+        lst: The list to slice
+        num_chunks: The amount of chunks that the list should be split into
+        chunk_number: Which chunk of the lst to return
+
+    Returns:
+        The chunk_number-th chunk of lst such that the concat of all num_chunks chunks is equivalent to the full lst,
+        and each chunk has equal size +-1
+    """
+    chunk_size = len(lst) // num_chunks
+    remainder = len(lst) % num_chunks
+    return lst[
+        chunk_number * chunk_size
+        + min(chunk_number, remainder) : (chunk_number + 1) * chunk_size
+        + int(chunk_number < remainder)
+        + min(chunk_number, remainder)
+    ]
+
+
+def replicate_datapoints(
+    client_src: CogniteClient,
+    client_dst: CogniteClient,
+    ts_external_id: str,
+    limit: Optional[int] = None,
+    partition_size: int = 100000,
+    mock_run: bool = False,
+    job_id: int = 1,
+):
+    """
+    Copies data points from the source tenant into the destination project, for the given time series.
+
+    If data points already exist in the destination for the time series, only the newer data points in the source are
+    copied over.
+
+    Args:
+        client_src: The client corresponding to the source project.
+        client_dst: The client corresponding to the destination project.
+        ts_external_id: The external id of the time series to replicate datapoints for
+        limit: The maximum number of data points to copy
+        partition_size: The maximum number of datapoints to retrieve per request
+        mock_run: If true, only retrieves data points from source and does not insert into destination
+        job_id: The batch number being processed
+    """
+    try:
+        latest_dst_dp = client_dst.datapoints.retrieve_latest(external_id=ts_external_id)
+        latest_src_dp = client_src.datapoints.retrieve_latest(external_id=ts_external_id)
+    except CogniteAPIError as exc:
+        logging.error(f"Job {job_id}: Failed for external id {ts_external_id}. {exc}")
+        return Replication.FAILURE, 0
+
+    start, end = _get_time_range(latest_src_dp, latest_dst_dp)
+    if not end:
+        return Replication.EMPTY, 0
+
+    logging.debug(f"Job {job_id}: Ext_id: {ts_external_id} Retrieving datapoints between {start} and {end}")
+    datapoints_count = 0
+    while start < end:
+        num_to_fetch = partition_size if limit is None else min(partition_size, limit - datapoints_count)
+        if num_to_fetch == 0:
+            break
+
+        try:
+            datapoints = client_src.datapoints.retrieve(
+                external_id=ts_external_id, start=start, end=end, limit=num_to_fetch
+            )
+            datapoints_count += len(datapoints)
+            if len(datapoints):
+                start = datapoints[-1].timestamp + 1

```

Comment:
line 98 and 100 can be moved outside the try/except statement, in an else statement?